src/main/java/org/jvnet/hudson/update_center/Plugin.java

```diff
@@ -455,7 +459,38 @@ public JSONObject toJSON() throws Exception {
 
         json.put("wiki", "https://plugins.jenkins.io/" + artifactId);
 
-        json.put("labels", getLabels());
+        GitHubSource gh = GitHubSource.getInstance();
+        if (scm != null && scm.contains("https://github.com/")) {
+            String[] parts = scm.replaceFirst("https://github.com/", "").split("/");
+            if (parts.length >= 2) {
+                String[] labels = gh.getTopics(parts[0], parts[1]).toArray(new String[0]);
+                if (labels.length > 0) {
+                    json.put("labels", labels);
+                }
+            }
+        }
+        if (json.has("labels")) {
+            json.put("labels", json.optJSONArray("labels").addAll(Arrays.asList(getLabels())));
+        } else {
+            json.put("labels", getLabels());
+        }
+
+        if (json.has("labels")) {
```

Comment:
When would this ever be false?

Reply:
if they have nothing defined in update center and nothing defined github?
I wrote the function after, so I guess they return an empty array, so its set to something and won't crash.